sklearn/neighbors/_base.py

```diff
@@ -346,6 +346,42 @@ def _check_algorithm_metric(self):
         if self.metric in ['wminkowski', 'minkowski'] and effective_p < 1:
             raise ValueError("p must be greater than one for minkowski metric")
 
+    def _check_metrics(self, other):
+        def _get_metric_params(obj):
+            params = (obj.metric_params.copy()
+                      if obj.metric_params is not None
+                      else {})
+            if 'p' not in params and hasattr(obj, 'p'):
+                params['p'] = obj.p
+            if obj.metric not in ['wminkowski', 'minkowski']:
+                del params['p']
```

Comment:
In case we have a custom metric taking a parameter `p`, we are heading into trouble here?